cpp/src/arrow/python/helpers.cc

```diff
@@ -321,6 +323,14 @@ void InitPandasStaticData() {
     pandas_NA = ref.obj();
   }
 
+  // Import DateOffset type
+  OwnedRef offsets;
+  if (internal::ImportModule("pandas.tseries.offsets", &offsets).ok()) {
```

Comment:
It seems that DateOffset if available in the top-level pandas namespace, in which case I would import it from there